# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/RTCRtpReceiver.cpp
# Commit: 3621f5bf1cb0
# Full Hash: 3621f5bf1cb07967a1b48d09d2b5f2cd99dfbc15
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2022-12-21 21:21:23
# Regressor Bug: 1800136
# File Overlap Count: 1
# Description:
#   Bug 1800136: Make RTCRtpSender::GetStats get ICE stats. r=ng
#   
#   Also fix bugs where we would get duplicate ICE stats. This mostly mattered to
#   WebrtcGlobalInformation since RTCStatsReportInternal does not automatically
#   remove duplicate ids like RTCStatsReport (RTCStatsReport is maplike).
# ==============================================================================

diff -r d2ace48b830c -r 3621f5bf1cb0 dom/media/webrtc/jsapi/RTCRtpReceiver.cpp
--- a/dom/media/webrtc/jsapi/RTCRtpReceiver.cpp	Tue Dec 20 22:12:09 2022 +0000
+++ b/dom/media/webrtc/jsapi/RTCRtpReceiver.cpp	Tue Dec 20 22:12:10 2022 +0000
@@ -176,7 +176,8 @@
   return promise.forget();
 }
 
-nsTArray<RefPtr<RTCStatsPromise>> RTCRtpReceiver::GetStatsInternal() {
+nsTArray<RefPtr<RTCStatsPromise>> RTCRtpReceiver::GetStatsInternal(
+    bool aSkipIceStats) {
   MOZ_ASSERT(NS_IsMainThread());
   nsTArray<RefPtr<RTCStatsPromise>> promises(3);
 
@@ -514,7 +515,7 @@
                                                         __func__);
               }));
 
-  if (GetJsepTransceiver().mTransport.mComponents) {
+  if (!aSkipIceStats && GetJsepTransceiver().mTransport.mComponents) {
     promises.AppendElement(mTransportHandler->GetIceStats(
         GetJsepTransceiver().mTransport.mTransportId,
         mPipeline->GetTimestampMaker().GetNow()));